src/module-loader.ts

```diff
@@ -99,17 +98,18 @@ export default class ModuleLoader {
    * the `modulePath` provided ends in `.mjs` it is assumed to be ESM.
    *
    * @param {string} filePath - File path to test.
+   * @param {IConfig|IPlugin} config - Oclif config or plugin config.
    *
    * @returns {boolean} The modulePath is an ES Module.
    * @see https://www.npmjs.com/package/get-package-type
    */
-  static isPathModule(filePath: string): boolean {
+  static isPathModule(filePath: string, config: IConfig|IPlugin): boolean {
     const extension = path.extname(filePath).toLowerCase()
 
     switch (extension) {
+    case '.ts':
     case '.js':
-      return getPackageType.sync(filePath) === 'module'
-
+      return config.pjson?.type === 'module'
```

Comment:
No need to have the `get-package-type` as the rule is:

### "commonjs" Behavior
`.cjs` files are treated as CommonJS modules
`.js` files are treated as CommonJS modules
`.mjs` files are treated as ES modules

### "module" Behavior
`.cjs` files are treated as CommonJS modules
`.js` files are treated as ES modules
`.mjs` files are treated as ES modules

So the difference on how `.js` files are treated only relies on the type field inside the `package.json`
As you can see, the difference is how .js files are treated.